### PR TITLE
Shalliburton/13688 validate document

### PR DESCRIPTION
**Pull request by sahalliburton**
Resolves [13688](https://github.com/department-of-veterans-affairs/caseflow/issues/13688)

### Description
Moved document ID validator from update legacy appeal workflow into a shared module and included in the updated legacy appeal workflow as well as Case Assignment. VACOLS case reviews can now use the same logic to check for a valid document ID as the legacy appeals.

### Acceptance Criteria
- [x] Code compiles correctly and tests pass
- [x] Acting Judge has one legacy appeal with Judge Task and one legacy appeal with Attorney Task

### Testing Plan
**NOTE:** You will need to reseed your DB to pick up the new legacy appeals
1. Become Acting Judge Kris
2. Navigate to the user's queue and verify there are 2 legacy appeals
3. Navigate to the first appeal and check the Document ID. 
- If it is NONE, verify the actions are related to an Attorney task. 
- If it contains an ID, verify that the actions are related to a Judge Task.
4. Navigate to the second appeal and repeat the steps above for the other Document ID
 
### Documentation Updates
- [x] Added description on seeding Case Reviews in FACOLS to [wiki](https://github.com/department-of-veterans-affairs/caseflow/wiki/FACOLS)

## Discovery Notes
https://hackmd.io/DsIbAObHRoeL7K_9BqqojA

### CODE REVIEW

File: lib/tasks/local/destroy.rake [hunk 1]
```diff
@@ -17,4 +17,38 @@ namespace :local do
 
     puts ">>> END local:destroy"
   end
+
+  desc "Nuke docker environment (removes all process, containers, images and volumes)"
```
Comment (pkarman) [inline lib/tasks/local/destroy.rake]: this seems cool but should it be in this PR?

Comment (sahalliburton) [reply lib/tasks/local/destroy.rake]: I can take it out of this PR, was using it to reset the environment for testing

File: app/workflows/update_legacy_attorney_case_review.rb [hunk 2]
```diff
@@ -28,6 +29,7 @@ def call
   private
 
   attr_reader :id, :document_id, :user, :success
+  delegate :work_product, to: :vacols_case_review
```
Comment (hschallhorn) [inline app/workflows/update_legacy_attorney_case_review.rb]: Why was this added?

File: db/seeds.rb [hunk 3]
```diff
@@ -1162,6 +1162,33 @@ def create_tasks_at_acting_judge
     create_task_at_judge_review(FactoryBot.create(:appeal), acting_judge, attorney)
   end
 
+  def create_legacy_appeal_at_acting_judge
+    # Find the 2 VACOLS Cases for the Acting Judge (seeded from local/vacols/VACOLS::Case_dump.csv)
```
Comment (hschallhorn) [inline db/seeds.rb]: I think we should include why "3662860" is assigned to this acting judge as an attorney and why "3662859" is assigned to this acting judge as judge. Perhaps something as simple as 
> Case 3662860 does not have a decision drafted for it yet, so it is assigned to the AVLJ as an attorney
> Case 3662859 has a valid decision document, so it is assigned to the AVLJ as a judge

File: db/seeds.rb [hunk 4]
```diff
@@ -1547,6 +1573,7 @@ def seed
     call_and_log_seed_step :create_inbox_messages
     call_and_log_seed_step :perform_seeding_jobs
     call_and_log_seed_step :setup_motion_to_vacate
+    call_and_log_seed_step :create_legacy_appeal_at_acting_judge
```
Comment (hschallhorn) [inline db/seeds.rb]: Would prefer to move this up to [here](https://github.com/department-of-veterans-affairs/caseflow/pull/13831/files#diff-f7a25bc30a409fc77050703fc048ca8cR1162) to keep all acting judge case creation together

File: spec/models/queues/legacy_work_queue_spec.rb [hunk 5]
```diff
@@ -26,16 +36,22 @@
...
+      it "returns attorney tasks for cases with a nil doc id, but judge task for cases with a valid doc id" do
+        expect(subject.length).to eq(3)
+        expect(subject[0].class).to eq(AttorneyLegacyTask)
+        expect(subject[1].class).to eq(JudgeLegacyDecisionReviewTask)
+      end
+
+      it "returns attorney tasks for cases with invalid doc ids, but judge task for cases with a valid doc id" do
+        expect(subject.length).to eq(3)
         expect(subject[0].class).to eq(AttorneyLegacyTask)
         expect(subject[1].class).to eq(JudgeLegacyDecisionReviewTask)
```
Comment (hschallhorn) [inline spec/models/queues/legacy_work_queue_spec.rb]: We add a third appeal, but never test what task type comes back for it. I think this can be left as one test
```ruby
      it "returns attorney tasks for cases with invalid doc ids, but judge task for cases with a valid doc id" do
        expect(subject.length).to eq(3)
        expect(subject[0].class).to eq(AttorneyLegacyTask)
        expect(subject[1].class).to eq(JudgeLegacyDecisionReviewTask)
        expect(subject[2].class).to eq(AttorneyLegacyTask)
```

File: app/workflows/update_legacy_attorney_case_review.rb (hunk 2 above)
Comment (sahalliburton) [reply app/workflows/update_legacy_attorney_case_review.rb:32]: When I refactored this originally, this line was a few lines down and so I was just moving it up. When I remove it, all of the `update_legacy_attorney_case_review_spec` tests fail. I believe this is because of the refactor into a module, it needs to know where to get work_product

File: app/validators/legacy_attorney_case_review_document_id_validator.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module LegacyAttorneyCaseReviewDocumentIdValidator
```
[the remark this answers is not in the document]
Comment (sahalliburton) [reply app/validators/legacy_attorney_case_review_document_id_validator.rb:3]: Thanks this was a lot of fun!

File: spec/models/queues/legacy_work_queue_spec.rb (hunk 5 above)
Comment (hschallhorn) [reply spec/models/queues/legacy_work_queue_spec.rb:50]: Looks like we still don't test that the appeal with a document_id of none "NONE" comes back as an AttorneyLegacyTask
